examples/statistics/time_series_histogram.py

```diff
@@ -0,0 +1,86 @@
+"""
+=====================
+Time Series Histogram
+=====================
+
+This example demonstrates how to efficiently visualize large numbers of time
+series in a way that could potentially reveal hidden substructure and patterns
+that are not immediately obvious.
+
+The first plot shows the typical way of visualizing multiple time series by
+overlaying them on top of each other with ``plt.plot``. The second and third
+plots show how to reinterpret the data as a 2d histogram.
+"""
+from copy import copy
+import time
+
+import numpy as np
+import numpy.matlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+_, axes = plt.subplots(nrows=3, figsize=(10, 6 * 3))
+
+# Make some data; a 1D random walk + small fraction of sine waves
+num_series = 10000
+num_points = 100
+SNR = 0.10  # Signal to Noise Ratio
+x = np.linspace(0, 4 * np.pi, num_points)
+# random walk
+Y = np.cumsum(np.random.randn(num_series, num_points), axis=-1)
+# sinusoidal signal
+num_signal = int(round(SNR * num_series))
+phi = (np.pi / 8) * np.random.randn(num_signal, 1)  # small random offest
+Y[-num_signal:] = ((
+    np.sqrt(np.arange(num_points))[None, :]
+    * np.sin(x[None, :] - phi))
+    + 0.05 * np.random.randn(num_signal, num_points)
+)
+
+# Plot it using `plot` and the lowest nonzero value of alpha (1/256).
+# With this view it is extremely difficult to observe the sinusoidal behavior
+# because of how many overlapping series there are. It also takes some time
+# to run because so many individual plots that need to be generated.
+tic = time.time()
+for i in range(Y.shape[0]):
+    axes[0].plot(x, Y[i], color="C0", alpha=1 / 256)
```

Comment:
Don't run this in a loop.  `plot(x, Y)` will work fine.

Reply:
You sure? I get an error `ValueError: x and y must have same first dimension, but have shapes (100,) and (10000, 100)`

Reply:
OK, can get it to work with
```python
X = np.broadcast_to(x[None, :], Y.shape)
axes[0].plot(X.T, Y.T, color="C0", alpha=1/256)
```

Reply:
also reduces plotting time by about 25%

Reply:
`axes[0].plot(x, Y.T, color="C0", alpha=1/256)`